### PR TITLE
Revive revive

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Reactivate the revive go linter for datadog-agent

### Motivation
revive was deactivated for a while and it brings some useful comments. Global deactivation of one rule doesn't work:
https://revive.run/docs#golangci-lint:
Please notice that if no particular configuration is provided, revive will behave as go-lint does, i.e. all go-lint rules are enabled (the [Available Rules table](https://revive.run/docs#available-rules) details what are the go-lint rules). When a configuration is provided, only rules in the configuration are enabled.

### Describe how to test/QA your changes
As a lot of modifications were done, it's safe to perform basic QA operations (start the agent and validate it sends basic telemetry is enough)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by FlorentClarret — APPROVED** [review]

File: test/fakeintake/client/client.go [hunk 1]
```diff
@@ -1,3 +1,7 @@
+// APIFakeIntakeRouteStatsGETResponse exported type should have comment or be unexported
+// RouteStat exported type should have comment or be unexported
+// APIFakeIntakePayloadsGETResponse exported type should have comment or be unexported
+// Payload exported type should have comment or be unexported
```
Comment (ogaca-dd) [inline test/fakeintake/client/client.go:4]: Why these changes?

**Review by ogaca-dd — APPROVED** [review]

**Review by paullegranddc — APPROVED** [review]

**Comment by guyarb** [comment]
Seeing the following changes which does not look good to put in the repo
image

File: test/fakeintake/client/client.go (hunk 1 above)
Comment (chouetz) [reply test/fakeintake/client/client.go:4]: good catch this is a corner case of my script I have missed

**Comment by chouetz** [comment]
> Seeing the following changes which does not look good to put in the repo image

@guyarb What should I do instead?

**Comment by guyarb** [comment]
> > Seeing the following changes which does not look good to put in the repo image
> 
> @guyarb What should I do instead?

I think the respective teams should make the changes on their own (until some date) and after that date the revive linter to back to life and be a required step in the agent

**Comment by guyarb** [comment]
BTW, why do we need to have a comment above the package? I don't think it will give any information beside

File: cmd/security-agent/subcommands/compliance/command.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func complianceEventCommand(globalParams *command.GlobalParams) *cobra.Command {
 				fx.Supply(eventArgs),
 				fx.Supply(core.BundleParams{
 					ConfigParams: config.NewSecurityAgentParams(globalParams.ConfigFilePaths),
-					LogParams:    log.LogForOneShot(command.LoggerName, "info", true),
+					LogParams:    log.ForOneShot(command.LoggerName, "info", true),
```
Comment (modernplumbing) [inline cmd/security-agent/subcommands/compliance/command.go:61]: Unfortunately we (security-agent) wouldn't have control over the naming of an exported func from another library: https://github.com/DataDog/datadog-agent/blob/main/comp/core/log/params.go#L61

I do like how revive provides pointers to how to write more nicely formatted code, but I think we may need to slowly up the strictness of it.

Comment (amenasria) [reply cmd/security-agent/subcommands/compliance/command.go:61]: It's a bit annoying to rename it everywhere, but @DataDog/agent-shared-components owns it and they accepted the PR so I guess it's fine ?

**Review by hmahmood — APPROVED** [review]

**Review by paullegranddc — APPROVED** [review]

File: cmd/security-agent/subcommands/compliance/command.go (hunk 2 above)
Comment (chouetz) [reply cmd/security-agent/subcommands/compliance/command.go:61]: I did change it as well :) I'm trying to fix the last small issues that remain on the PR and will dispatch the PR differently to all teams

**Review by brycekahle — APPROVED** [review]
